### ISSUE TITLE
-b option creates odd layout

### ISSUE THREAD

**Issue by rsmacleod** [SCIInstitute/map3d#21]
When I use the -b option to create a single window with sub window content, the top piece of the resulting window has empty white space with some geometry file names listed at the top and then the contents organized near the bottom.  The test scripts show the behavior very well as does the attached screen shot.
![map3dscreensnapz001](https://cloud.githubusercontent.com/assets/2551863/16923931/fa57794e-4cd9-11e6-8127-7a4496c84882.png)

**Comment by bryanworthen** [comment]
Okay. This shouldn't be too hard.

**Comment by bryanworthen** [comment]
Can you attach your script? I think you have some -as, -al, -at options that determine positions of things.

**Comment by rsmacleod** [comment]
HI Bryan.  This is one of the original test scripts we included with the program.  Github will not allow me to attach it (another of my points against Github (-: ) You will need to define a location for map3d but the rest of script should run on a Mac or Unix. 

Cheers,
Rob

# This one shows two different data sets on the same geometry

# Note the use of MATLAB files that contain multiple time series.

# There is also explicit control of the time instants to read

# and display; this aligns the two signals for better comparison.

# Also the use of locked scaling and the supression of one of the legend

# windows.

# See if you can spot the difference in the two time sequences.

# Hint: look in the late ST segment and especially the T wave.

#######################################################################

#MAP3D=../map3d
MAP3D=/Applications/map3d.app/Contents/MacOS/map3d
GEOM=../geom/tank
DATA=../data/tank

$MAP3D -nw -b -f ${GEOM}/25feb97_sock_closed.fac \
    -as 200 600 400 800 \
    -p ${DATA}/cool1-atdr.mat -s 82 420 \
    -sl 2 \
    -at 200 600 250 400 -t 126\
    -al 1 200 500 750 \
    -ch ${GEOM}/sock128.channels \
    -lm ${GEOM}/25feb97_sock.lmarks \
    -f ${GEOM}/25feb97_sock_closed.fac \
    -as 590 990 400 800 \
    -p ${DATA}/cool1-atdr.mat -s 74 412 \
    -slw 0 \
    -at 590 990 250 400 -t 126 \
        -al 1 200 250 500  \
    -ch ${GEOM}/sock128.channels \
    -lm ${GEOM}/25feb97_sock.lmarks

**Comment by bryanworthen** [comment]
Yes, I hate that about github too, but I think that you can do zip files.

Thanks,
Bryan

**Comment by bryanworthen** [comment]
Okay, I ran it and came up with a similar result.  However, from the past, I think it is working as intended:

The script is going out of its way to define the subwindows at 250, so the WindowManager is reserving 250 pixels of space between the geometry file names.

It sounds like you would rather have it detect the empty space and reduce it.  Thus, when you make absolute positions in borderless mode, they really serve to be "relative" positions if you don't flush them at 1.

I can't tell if you have an issue with the displaying of geometry file names.  I remember doing something like this - it makes room for up to 3 file names, and will add "and others" if there are more.  It feels like they are kind of getting in the way now.

Let me know your thoughts,
Bryan

**Comment by rsmacleod** [comment]
OK, the problem is that the documentation claims the origin for placing windows is the **lower** left corner of the window but it appears to be the *_upper
![map3dscreensnapz002](https://cloud.githubusercontent.com/assets/2551863/17201299/758fe8c2-544a-11e6-9d62-ea6e43d49842.png)
*_ left corner of the window. With that correction in assumptions, most of the following script works:

MAP3D=/Applications/map3d.app/Contents/MacOS/map3d
GEOM=../geom/tank
DATA=../data/tank

$MAP3D -nw -b -f ${GEOM}/25feb97_sock_closed.fac \
    -as 200 600 150 500 \
    -p ${DATA}/cool1-atdr.mat -s 82 420 \
    -sl 2 \
    -at 200 600 1 149 -t 126\
    -al 1 200 1 250 \
    -ch ${GEOM}/sock128.channels \
    -lm ${GEOM}/25feb97_sock.lmarks \
    -f ${GEOM}/25feb97_sock_closed.fac \
    -as 590 990 150 500  \
    -p ${DATA}/cool1-atdr.mat -s 74 412 \
    -slw 0 \
    -at 590 990 1 149 -t 126 \
        -al 1 200 250 499 \
    -ch ${GEOM}/sock128.channels \
    -lm ${GEOM}/25feb97_sock.lmarks 

However, the second legend bar is not visible, the one that should sit below the first. Also, there is some overlap of the labels, the Surface # 1 label from the surface is cut off by the Node # label from the time signal, even though the windows should not overlap.  One ends at 149 and the next begins at 150.  Similarly in the time series window, the Value label is cut off, does not fit in the window. And no amount of font scaling seems to fix this.
Finally, is there a mouse-click combo that allows the user to move the sub windows around within the big window or to resize them?  This is a handy feature for fixing the layout problems and I thought there used to be a way to do this.
Thanks!
Rob
